### PR TITLE
Park styling

### PR DESCRIPTION
**Pull request by Candide23**
styling and login

### CODE REVIEW

File: src/main/java/com/liftoff/parksnrec/Controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,2 @@
+package com.liftoff.parksnrec.Controller;public class UserController {
```
Comment (jaylingelbach) [inline src/main/java/com/liftoff/parksnrec/Controller/UserController.java:1]: Please remove all files/folders not related to the UI styling story.

File: src/main/java/com/liftoff/parksnrec/Model/User.java [hunk 2]
```diff
@@ -0,0 +1,2 @@
+package com.liftoff.parksnrec.Model;public class User {
```
Comment (jaylingelbach) [inline src/main/java/com/liftoff/parksnrec/Model/User.java:1]: Please remove all files/folders not related to the UI styling story.

File: src/main/java/com/liftoff/parksnrec/Repository/UserRepository.java [hunk 3]
```diff
@@ -0,0 +1,2 @@
+package com.liftoff.parksnrec.Repository;public class UserRepository {
```
Comment (jaylingelbach) [inline src/main/java/com/liftoff/parksnrec/Repository/UserRepository.java:1]: Please remove all files/folders not related to the UI styling story.

File: src/main/ui/package.json [hunk 4]
```diff
@@ -1,56 +1,17 @@
...
-  },
-
 	"name": "parks-n-rec",
 	"version": "0.1.0",
 	"private": true,
 	"dependencies": {
 		"@testing-library/jest-dom": "^5.16.4",
 		"@testing-library/react": "^13.3.0",
 		"@testing-library/user-event": "^13.5.0",
+		"axios": "^0.27.2",
```
Comment (jaylingelbach) [inline src/main/ui/package.json:9]: I don't want this dependency, should be using `fetch` which is built in to JS

File: src/main/ui/package.json [hunk 5]
```diff
@@ -1,56 +1,17 @@
...
 		"bootstrap": "^5.2.0",
 		"react": "^18.2.0",
 		"react-bootstrap": "^2.5.0",
 		"react-dom": "^18.2.0",
+		"react-router-dom": "^6.3.0",
```
Comment (jaylingelbach) [inline src/main/ui/package.json:14]: We don't need a router. Please remove.

File: src/main/ui/src/App.js [hunk 6]
```diff
@@ -1,12 +1,23 @@
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'
```
Comment (jaylingelbach) [inline src/main/ui/src/App.js:1]: Please remove all routing. This is not related to the story you are working on.

File: src/main/ui/src/components/Login/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+import basestyle from "./Base.module.css";
+import loginstyle from "./Login.module.css";
+import axios from "axios";
+import { useNavigate, NavLink } from "react-router-dom";
+const Login = ({ setUserState }) => {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx:6]: please change to the agreed upon format 

`export default function <function-name>() {

}


// no export needed at the bottom

File: src/main/ui/src/components/Login/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      error.password = "Password is required";
+    }
+    return error;
+  };
+
+  const loginHandler = (e) => {
+    e.preventDefault();
+    setFormErrors(validateForm(user));
+    setIsSubmit(true);
+    // if (!formErrors) {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx:43]: remove unused comments please :)

File: src/main/ui/src/components/Login/Login.jsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+    // }
+  };
+
+  useEffect(() => {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      console.log(user);
+      axios.post("http://localhost:3000/login", user).then((res) => {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx:51]: convert to fetch

File: src/main/ui/src/components/Login/Login.jsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+        alert(res.data.message);
+        setUserState(res.data.user);
+        navigate("/", { replace: true });
+      });
+    }
+  }, [formErrors]);
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx:57]: getting error in VSCode

`const formErrors: {}
React Hook useEffect has missing dependencies: 'isSubmit', 'navigate', 'setUserState', and 'user'. Either include them or remove the dependency array. If 'setUserState' changes too often, find the parent component that defines it and wrap that definition in useCallback`

File: src/main/ui/src/components/Header/Button.css [hunk 11]
```diff
@@ -0,0 +1,12 @@
+.btn-primary {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Header/Button.css:1]: Let's remove this and use the bootstrap styling that will match emily's buttons.
Screen Shot 2022-09-02 at 11 31 23 AM

Also it doesn't line up. Maybe that will be fixed when using the default bootstrap?

File: src/main/ui/src/components/Header/Header.jsx [hunk 12]
```diff
@@ -3,6 +3,11 @@ import Nav from "react-bootstrap/Nav";
 import Navbar from "react-bootstrap/Navbar";
 import SearchBar from "../SearchBar/SearchBar";
 import StateDropdown from "./StateDropdown";
+import Button from 'react-bootstrap/Button';
+import './Button.css'
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Header/Header.jsx:7]: remove and use bootstrap button already imported.

File: src/main/ui/src/components/Login/Base.module.css [hunk 13]
```diff
@@ -0,0 +1,17 @@
+.button_common {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Base.module.css:1]: I think we could use the bootstrap button and it's built in errors here as well.

File: src/main/ui/src/components/Login/Login.jsx [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+    // if (!formErrors) {
+
+    // }
+  };
+
+  useEffect(() => {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      console.log(user);
+      axios.post("http://localhost:3000/login", user).then((res) => {
+        alert(res.data.message);
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx:52]: No alerts. It interrupts the user experience and stops all functions on a SPA app. can log errors to the console or if it's being relayed to the user by a css error class. Bootstrap should be able to help if it's an error the user should see.

File: src/main/ui/src/components/Login/Login.jsx [hunk 15]
```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+import basestyle from "./Base.module.css";
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx:2]: if this is used across the app you can refactor this into the app.css file or whatever file is being injected throughout the app. you can just add the style code in that file

File: src/main/ui/src/components/Login/Register.jsx [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+    } else if (values.veryPassword !== values.password) {
+      error.veryPassword = "Confirm password and password should be same";
+    }
+    return error;
+  };
+  const signupHandler = (e) => {
+    e.preventDefault();
+    setFormErrors(validateForm(user));
+    setIsSubmit(true);
+    // if (!formErrors) {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Register.jsx:53]: remove unused comments

File: src/main/ui/src/components/Login/Register.jsx [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+    //   setIsSubmit(true);
+    // }
+  };
+
+  useEffect(() => {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      console.log(user);
+      axios.post("http://localhost:3000/signup/", user).then((res) => {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Register.jsx:61]: refactor to fetch

File: src/main/ui/src/components/Login/Register.jsx [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+        alert(res.data.message);
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Register.jsx:62]: no alerts.

File: src/main/ui/src/components/Login/Register.jsx [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+        navigate("/login", { replace: true });
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Register.jsx:63]: when we navigate to login I do not want to still see the list of parks.

Comment (jaylingelbach) [reply src/main/ui/src/components/Login/Register.jsx:63]: same with register.

File: src/main/ui/src/components/Login/Register.module.css [hunk 20]
```diff
@@ -0,0 +1,30 @@
+.register {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Register.module.css:1]: your css is great. looks great, responsive. Nice work!

File: src/main/ui/src/components/SearchByName/SearchByName.jsx [hunk 21]
```diff
@@ -1,5 +1,10 @@
 import React from 'react';
```
Comment (jaylingelbach) [inline src/main/ui/src/components/SearchByName/SearchByName.jsx:1]: Let's take the `searchByName` changes out. It's still broken and displaying duplicates. We can make the work a different branch that you can tackle next. But make the changes on everything else and we can then move to styling `searchByName`.

**Review by jaylingelbach — CHANGES_REQUESTED** [review]
Styling is great! Good work. Have some notes on things, this is all normal stuff. Keep up the good work!!!!

Also, make sure you fix any linting errors before submitting and an errors in the console of the browser. But overall fantastic work!
Screen Shot 2022-09-02 at 11 51 52 AM

**Comment by jaylingelbach** [comment]
Screen Shot 2022-09-02 at 11 53 47 AM

File: src/main/ui/src/components/Header/Header.jsx [hunk 22]
```diff
@@ -3,6 +3,11 @@ import Nav from "react-bootstrap/Nav";
 import Navbar from "react-bootstrap/Navbar";
 import SearchBar from "../SearchBar/SearchBar";
 import StateDropdown from "./StateDropdown";
+import Button from 'react-bootstrap/Button';
+
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Header/Header.jsx:7]: remove all this space please. Doesn't look as if you install the linter like I asked a while ago. I'll DM you the link to setup

File: src/main/ui/src/components/Header/Header.jsx [hunk 23]
```diff
@@ -15,8 +20,13 @@ export const Header = () => {
 						<SearchBar />
 						<StateDropdown />
 					</Nav>
-				</Navbar.Collapse>
+					<div style={{ display: "block", height: 49 }}>
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Header/Header.jsx]: for now run all the code through jsfiddle.com in the javascript section there is a button in the top right corner of the JS part that says tidy.

File: src/main/ui/src/components/Login/Login.jsx [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+
+export default function Login() {
+
+       const [password,setPassword]=useState('')
+       const [email,setEmail]=useState('')
+
+
+
+       const handleClick=(e)=>{
+           e.preventDefault()
```
Comment (jaylingelbach) [inline src/main/ui/src/components/Login/Login.jsx]: remove, but save for later, we don't have the backend implemented yet.

**Review by jaylingelbach — CHANGES_REQUESTED** [review]
To avoid any confusion i've made some changes to your code. 

Do a git pull.

I removed all the extra css files that we specific to the components and moved the styling to the app.css file.

You will need to use bootstrap to keep styling consistent for buttons.
I have removed any button styling from your code. 
They have built in error styling as well that you can adapt for your forms.

I have removed any routing as that is a whole other bag of worms that we need to work through. I included your components in the app.js file so they can be seen.
You can rework the styles from there, but keep in mind bootstrap has error styling built in so you won't need your custom error styles.

I'm probably forgetting something so if you have questions, reach out. Also let me know each commit you make so I can make sure we are going in the right direction.

Thanks!!!!
Jay

File: src/main/ui/src/components/SearchByName/SearchByName.jsx [hunk 25]
```diff
@@ -29,38 +34,51 @@ export default function SearchByName() {
...
+                <div key={key}>
+                  <Row xs={2} md={2} className="g-4" >
 
-     <div>
-             <input
-                 type="text"
-                 placeholder="Search parks by name"
-                 onChange = {handleSearch}
-             />
+                       {Array.from({ length: 2 }).map((_, idx) => (
```
Comment (jaylingelbach) [inline src/main/ui/src/components/SearchByName/SearchByName.jsx:61]: This is what is making the duplicates. Change to `length: 1` and adjust so each row has two items in it. RIght now it looks like these are arranged with two columns instead of one col and two items per row.

**Review by jaylingelbach — CHANGES_REQUESTED** [review]

File: src/main/ui/src/components/SearchByName/SearchByName.jsx (hunk 25 above)
Comment (Candide23) [reply src/main/ui/src/components/SearchByName/SearchByName.jsx]: Emily told me she can resolve that I will talk to her today

File: src/main/ui/src/components/Header/Header.jsx [hunk 26]
```diff
@@ -3,6 +3,8 @@ import Nav from "react-bootstrap/Nav";
 import Navbar from "react-bootstrap/Navbar";
```
[the remark this answers is not in the document]
Comment (Candide23) [reply src/main/ui/src/components/Header/Header.jsx]: done

**Review by jaylingelbach — APPROVED** [review]
